grpclb/src/main/java/io/grpc/grpclb/GrpclbClientLoadRecorder.java

```diff
@@ -83,52 +100,55 @@ ClientStats generateLoadReport() {
     ClientStats.Builder statsBuilder =
         ClientStats.newBuilder()
         .setTimestamp(Timestamps.fromMillis(time.currentTimeMillis()))
-        .setNumCallsStarted(callsStarted.getAndSet(0))
-        .setNumCallsFinished(callsFinished.getAndSet(0))
-        .setNumCallsFinishedWithClientFailedToSend(callsFailedToSend.getAndSet(0))
-        .setNumCallsFinishedKnownReceived(callsFinishedKnownReceived.getAndSet(0));
-    HashMap<String, AtomicLong> savedCallsDroppedPerToken;
+        .setNumCallsStarted(callsStartedUpdater.getAndSet(this, 0))
+        .setNumCallsFinished(callsFinishedUpdater.getAndSet(this, 0))
+        .setNumCallsFinishedWithClientFailedToSend(callsFailedToSendUpdater.getAndSet(this, 0))
+        .setNumCallsFinishedKnownReceived(callsFinishedKnownReceivedUpdater.getAndSet(this, 0));
+
+    Map<String, LongHolder> localCallsDroppedPerToken = Collections.emptyMap();
     synchronized (this) {
-      savedCallsDroppedPerToken = callsDroppedPerToken;
-      callsDroppedPerToken = new HashMap<String, AtomicLong>();
+      if (!callsDroppedPerToken.isEmpty()) {
```

Comment:
In a noisy CL like this, a change like this which changes the logic is _really_ hard to notice. It should be separate or called out for reviewers.

This change itself looks fine, but it needs to be seen by reviewers.

Reply:
Fair enough.  I was hoping that the behavior was identical, which is why I didn't bring it up.